src/plugins/discover/public/context_awareness/profile_providers/observability/logs_data_source_profile/profile.ts

```diff
@@ -19,13 +19,17 @@ import { extractIndexPatternFrom } from '../../extract_index_pattern_from';
 export const createLogsDataSourceProfileProvider = (
   services: ProfileProviderServices
 ): DataSourceProfileProvider => ({
-  profileId: 'logs-data-source-profile',
+  profileId: 'observability-logs-data-source-profile',
   profile: {
     getCellRenderers,
     getRowIndicatorProvider,
     getRowAdditionalLeadingControls,
   },
   resolve: (params) => {
+    if (params.rootContext.solutionType !== SolutionType.Observability) {
+      return { isMatch: false };
+    }
```

Comment:
> [!TIP]
> Non-blocking suggestion, as this would work perfectly 👌 
>
> This condition should probably reflect how we resolve the O11y root profile, which means this needs to change and be aligned every time we update the logic for the O11y root profile resolution.
>
> I don't expect that to change often, but we might miss some corner cases if we forget to update the logic also here and anywhere else this is implemented in this PR.
>
> Have you considered to use the root profile resolve declaration to keep the resolution logic centralized and inherit it here?
```ts
resolve: async (params) => {
  const observabilityRootProfile = await createObservabilityRootProfileProvider(services).resolve(
    { solutionNavId: params.rootContext.solutionType }
  );

  if (!observabilityRootProfile.isMatch) {
    return { isMatch: false };
  }
  
  // ...
},
```

Reply:
As we discussed offline, while we can't re-execute the root profile resolution logic (since it's async while document profile resolution is sync), I've updated the `rootContext` checks to look at `profileId` instead of `solutionType`, which has the same effect of protecting against future changes: https://github.com/elastic/kibana/pull/199255/commits/a41d43ebcb42fdc7f25982337b044250cdcd4637.